### PR TITLE
Docs: various updates

### PR DESCRIPTION
**Pull request by bpierre**

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -1,15 +1,18 @@
 # Table of contents
 
-* [Aragon Govern](README.md)
+* [Introduction](README.md)
 
-## High level overview <a id="guides"></a>
+## Overview <a id="guides"></a>
```
Comment (bpierre) [inline docs/SUMMARY.md:5]: I think using “Overview” is enough and aligns more with the other section name (“Developers”).

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 * [ERC3000](guides/erc3000.md)
-* [Govern Core concepts](guides/core-concepts.md)
+* [Core concepts](guides/core-concepts.md)
 
 ## Developers
 
-* [Getting started](developers/quick-start.md)
-* [Smart contracts breakdown](developers/getting-started.md)
+* [Getting started](developers/getting-started.md)
+* [Smart contracts breakdown](developers/smart-contracts.md)
```
Comment (bpierre) [inline docs/SUMMARY.md:13]: @Evalir I renamed these files to align them with their titles.

**Review by bpierre — COMMENTED** [review]
@Evalir I created some sections in the getting started rather than having the list of packages, so we can have visuals and group similar packages together. Let me know what you think!

**Review by Evalir — APPROVED** [review]
✨

File: docs/SUMMARY.md (hunk 1 above)
Comment (Evalir) [reply docs/SUMMARY.md:5]: Fully agree, it's way better!

**Pull request merged**